core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
+      _rddInfoMap.get(info.id) match {
+        case Some(rddInfo) if info.name != rddInfo.name =>
+          rddInfo.name = info.name
+        case None =>
+          _rddInfoMap(info.id) = info
+        case _ =>
```

Comment:
we don't need `case _ =>`. If you use @zsxwing's suggestion above then this comment doesn't matter anymore.